### PR TITLE
tuxpaint: add mainProgram and use finalAttrs

### PR DESCRIPTION
**Pull request by peterhoeg**
## Description of changes

as the title says

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/games/tuxpaint/default.nix [hunk 1]
```diff
@@ -28,13 +28,14 @@ let
     url = "mirror://sourceforge/project/tuxpaint/tuxpaint-stamps/2024-01-29/tuxpaint-stamps-2024.01.29.tar.gz";
     hash = "sha256-GwJx9tqaX7I623tJQYO53iiaApZtYsTLQw2ptBIFlKk=";
   };
+
```
Comment (Aleksanaa) [inline pkgs/games/tuxpaint/default.nix:31]: So you added a blank line

File: pkgs/games/tuxpaint/default.nix [hunk 2]
```diff
@@ -28,13 +28,14 @@ let
...
 in
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
```
Comment (Aleksanaa) [inline pkgs/games/tuxpaint/default.nix:33]: and `finalAttrs` 🙃

**Review by drupol — CHANGES_REQUESTED** [review]
The PR title is misleading. Can you please split this in 2 commits (finalAttrs and mainProgram) and update the PR title accordingly? Thanks.

**Comment by peterhoeg** [comment]
It's slightly misleading, correct, but splitting up the commits seems a little excessive. I have a hard time imagining a scenario, where we would want to revert only one of the changes. They only affect things at build-time, so any issues would appear then.

**Comment by drupol** [comment]
I also share the same point of view, but this is how it's usually done in the project. I got my own PRs rejected because of exactly the same thing.

**Comment by peterhoeg** [comment]
The thing is, we really lack standards in the project - one person might say "you *must* fix this" while somebody is perfectly fine with it. 

So if you're cool with it and I'm cool with it, I'd say unless the maintainer of this package complains, let's just go ahead and commit this so we can move forward.

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
Right, let's move on.

**Pull request merged**